org.eclipse.scanning.event.ui/src/org/eclipse/scanning/event/ui/view/DefaultScanResultsHandler.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.scanning.event.ui.view;
+
+import java.io.File;
+
+import org.eclipse.core.filesystem.EFS;
+import org.eclipse.core.filesystem.IFileStore;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.scanning.api.event.scan.ScanBean;
+import org.eclipse.scanning.api.event.status.StatusBean;
+import org.eclipse.scanning.api.ui.IResultHandler;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.IEditorDescriptor;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IEditorRegistry;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.FileStoreEditorInput;
+
+/**
+ * Result handler for opening {@link ScanBean}s by opening their results files, as
+ * returned by {@link ScanBean#getFilePath()}.
+ * 
+ * @author Matthew Dickie
+ */
+public class DefaultScanResultsHandler implements IResultHandler<ScanBean> {
+
+	private static final String DATA_PERSPECTIVE_ID = "org.edna.workbench.application.perspective.DataPerspective";
+
+	@Override
+	public boolean isHandled(StatusBean bean) {
+		return bean.getRunDirectory() == null &&
+				(bean instanceof ScanBean && ((ScanBean) bean).getFilePath() != null);
+	}
+
+	@Override
+	public boolean open(ScanBean scanBean) throws Exception {
+		if (scanBean.getFilePath() == null) {
+			return false;
+		}
+		
+		if (!scanBean.getStatus().isFinal() && !confirmOpen(scanBean)) {
+			return false;
+		}
+		
+		String filePath = scanBean.getFilePath();
+		IWorkbench workbench = PlatformUI.getWorkbench();
+		IWorkbenchWindow window = workbench.getActiveWorkbenchWindow();
+
+		// Set the perspective to Data Browsing Perspective
+		// TODO FIXME When there is a general data viewing perspective from DAWN, use that.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1134)
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)